core/src/main/java/com/orientechnologies/orient/core/storage/cache/local/twoq/O2QCache.java

```diff
@@ -380,7 +381,7 @@ private UpdateCacheResult doLoad(long fileId, long pageIndex, boolean checkPinne
   }
 
   @Override
-  public OCacheEntry allocateNewPage(long fileId, OWriteCache writeCache) throws IOException {
+  public OCacheEntry allocateNewPage(long fileId, OWriteCache writeCache, boolean verifyChecksums) throws IOException {
```

Comment:
AFAIU, the underlying read/write cache API does not actually require the page to be really a new page. If the requested `filledUpTo` page already exists it's just loaded from disk and passed as a newly allocated one back to the caller of `allocateNewPage`. I think it's up to the caller to decide whether he wants to perform the checksum verification or not in such case.